plugins/modules/inventory_workflow_manager.py

```diff
@@ -1788,6 +1932,34 @@ def provisioned_wired_device(self):
 
         return self
 
+    # Added function
+    def assign_device_to_site(self, device_ip, assign_network_device_to_site):
+        try:
+            self.log(assign_network_device_to_site)
+            response = self.dnac._exec(
+                family="site_design",
+                function='assign_network_devices_to_a_site',
+                op_modifies=True,
+                params=assign_network_device_to_site,
+            )
+            self.log("Received API response from 'assign_network_devices_to_a_site': {0}".format(str(response)), "DEBUG")
+            task_id = response.get("response").get("taskId")
+            self.log(task_id)
+
+            while True:
```

Comment:
This will be infinite while loop.. Can you check it? We already have some APIs to do it... 

```
  # Monitor the task progress with a retry limit
        retry_limit = 10. <<< Check our dnac retry limit
        retry_count = 0. 
        while retry_count < retry_limit:
            execution_details = self.get_task_details(task_id)
            progress = execution_details.get("progress", "")
            is_error = execution_details.get("isError", False)

            if "success" in progress:
                self.log("Successfully assigned site to the device {0}".format(device_ip))
                return True
            elif is_error:
                error_msg = "Assigning site failed for device {0}. Error details: {1}".format(device_ip, execution_details)
                self.log(error_msg, "ERROR")
                return False

            # Wait before retrying to check the task status again
            time.sleep(sleep_interval)
            retry_count += 1
```

Reply:
updated the code